modules/articles/client/controllers/articles.client.controller.js

```diff
@@ -1,84 +1,52 @@
-'use strict';
+(function () {
+  'use strict';
 
-// Articles controller
-angular.module('articles').controller('ArticlesController', ['$scope', '$stateParams', '$location', 'Authentication', 'Articles',
-  function ($scope, $stateParams, $location, Authentication, Articles) {
-    $scope.authentication = Authentication;
+  angular
+    .module('articles')
+    .controller('ArticlesController', ArticlesController);
 
-    // Create new Article
-    $scope.create = function (isValid) {
-      $scope.error = null;
+  ArticlesController.$inject = ['$scope', '$state', 'articleResolve', 'Authentication'];
 
-      if (!isValid) {
-        $scope.$broadcast('show-errors-check-validity', 'articleForm');
-
-        return false;
-      }
-
-      // Create new Article object
-      var article = new Articles({
-        title: this.title,
-        content: this.content
-      });
+  function ArticlesController($scope, $state, article, Authentication) {
+    var vm = this;
 
-      // Redirect after save
-      article.$save(function (response) {
-        $location.path('articles/' + response._id);
-
-        // Clear form fields
-        $scope.title = '';
-        $scope.content = '';
-      }, function (errorResponse) {
-        $scope.error = errorResponse.data.message;
-      });
-    };
+    vm.article = article;
+    vm.authentication = Authentication;
+    vm.error = null;
+    vm.form = {};
+    vm.remove = remove;
```

Comment:
Ref: [Bindable Members Up Top](https://github.com/johnpapa/angular-styleguide#style-y033)

Setting anonymous functions in-line can be easy, but when those functions are more than 1 line of code they can reduce the readability. Defining the functions below the bindable members (the functions will be hoisted) moves the implementation details down, keeps the bindable members up top, and makes it easier to read.